### PR TITLE
[windows] Fix driver byte counts

### PR DESCRIPTION
**Pull request by derekwbrown**
Incorporates new driver which fixes byte counts when TCP 4 tuple is reused quickly.

Also adds fixes in both driver and ETW for zero latency http requests.

Adds new stats for observing conditions that lead to zero latency and lead to the byte overcounts

### Motivation

Customer ticket showed problem with byte counts
### Describe how to test/QA your changes

Load NPM test environment with high load connection counts (>2000cps).  Prior to fix, note that there would be unreasonably
high byte counts.  Now note expected byte counts.

### Reviewer's Checklist
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by richl-dd — APPROVED** [review]

File: releasenotes/notes/fixbyteovercount-71db480b4025b60d.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fixes bug on Windows in which NPM displays byte overcounts.
```
Comment (maycmlee) [inline releasenotes/notes/fixbyteovercount-71db480b4025b60d.yaml:11]: Is it correct to say "Windows driver"?

```suggestion
    Fixes a bug on the Windows driver where NPM displays byte overcounts.
```

File: pkg/network/state.go [hunk 2]
```diff
@@ -188,6 +188,7 @@ type networkState struct {
 
 // NewState creates a new network state
 func NewState(clientExpiry time.Duration, maxClosedConns uint32, maxClientStats int, maxDNSStats int, maxHTTPStats int, maxKafkaStats int) State {
+
```
Comment (hmahmood) [inline pkg/network/state.go:191]: Looks like this is the only change that requires a review from networks.

File: releasenotes/notes/fixbyteovercount-71db480b4025b60d.yaml (hunk 1 above)
Comment (derekwbrown) [reply releasenotes/notes/fixbyteovercount-71db480b4025b60d.yaml:11]: ok... but then how about
`Fixes a bug in the Windows NPM driver where NPM displays byte overcounts.

It reads a bit redundant, but there is more than one driver on windows.

**Review by dustmop — APPROVED** [review]
LGTM for agent-shared-components

**Review by vickenty — APPROVED** [review]

**Review by bkabrda — APPROVED** [review]

**Review by maycmlee — APPROVED** [review]

**Pull request merged**